app/src/main/java/com/example/checkpoint/services/WeatherService.kt

```diff
@@ -8,8 +8,12 @@ import kotlinx.coroutines.async
 import kotlinx.coroutines.withContext
 import retrofit2.awaitResponse
 
-class WeatherService {
-    suspend fun fetchWeather(): WeatherAPI{
+interface IWeatherService {
+    suspend fun fetchWeather(): WeatherAPI
+}
+
```

Comment:
SOLID Principle D Dependency Inversion states that you should implement and interface before a direct class.
https://www.digitalocean.com/community/conceptual_articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design#dependency-inversion-principle